backend/graphql/resolvers/courseResolvers.ts

```diff
@@ -16,14 +20,70 @@ import UserService from "../../services/implementations/userService";
 import IEmailService from "../../services/interfaces/emailService";
 import IUserService from "../../services/interfaces/userService";
 import { Role } from "../../types";
-import { CourseVisibilityAttributes } from "../../models/course.model";
+import { CourseVisibilityAttributes, Module } from "../../models/course.model";
 import { assertNever } from "../../utilities/errorUtils";
 
 const courseService: ICourseService = new CourseService();
 const userService: IUserService = new UserService();
 const emailService: IEmailService = new EmailService(nodemailerConfig);
 const authService: IAuthService = new AuthService(userService, emailService);
 
+function serializeToModuleDTO(modules: { [id: string]: Module }): ModuleDTO[] {
+  return Object.entries(modules).map(([key, val]) => ({ id: key, ...val }));
+}
+
+function deserializeModuleDTO(
+  moduleDTOs: ModuleDTO[],
+): { [id: string]: Module } {
+  const modules: { [id: string]: Module } = {} as { [id: string]: Module };
+
+  moduleDTOs.forEach((moduleDTO) => {
+    const moduleObj = {
+      title: moduleDTO.title,
+      description: moduleDTO.description,
+      image: moduleDTO.image,
+      previewImage: moduleDTO.previewImage,
+      published: moduleDTO.published,
+      lessons: moduleDTO.lessons,
+    };
+
+    // If no UUID is defined, create one
+    modules[moduleDTO?.id ?? uuidv4()] = moduleObj;
+  });
+
+  return modules;
+}
+
+function serializeCourseResponse(
+  course: CourseResponseDTO,
+): SerializedCourseResponseDTO {
+  if (course?.modules) {
+    return {
+      ...course,
+      modules: serializeToModuleDTO(course.modules),
+    };
+  }
+  return {
+    ...course,
+    modules: [],
+  };
+}
+
+function deserializeCourseRequest(
+  course: SerializedCreateCourseRequestDTO | SerializedUpdateCourseRequestDTO,
+): CreateCourseRequestDTO | UpdateCourseRequestDTO {
+  if (course?.modules) {
+    return {
+      ...course,
+      modules: deserializeModuleDTO(course.modules),
+    };
+  }
+  return {
+    ...course,
+    modules: {},
+  };
+}
```

Comment:
same comments as on `serialize` (less verbose version, unexpected usages of `?`)